src/modm/processing/fiber/fiber.hpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2020, Erik Henriksson
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#pragma once
+
+#include "context.hpp"
+#include <modm/debug/logger.hpp>
+
+#include <functional>
+
+namespace modm {
+
+class Fiber;
+modm_always_inline void yield();
+
+namespace fiber {
+
+class Scheduler;
+
+template<int size>
+class Stack {
+  static_assert(size % 4 == 0, "Stack size must be multiple of 4.");
```

Comment:
of 8. The AAPCS expect the stack to be 8B aligned when it links to functions outside of the current compilation unit. Don't see a technical reason for this tho, I suspect some reuse of Cortex-A compiler code.